subprojects/tooling-api/src/main/java/org/gradle/tooling/model/idea/IdeaContentRoot.java

```diff
@@ -58,6 +58,29 @@
     @Incubating
     DomainObjectSet<? extends IdeaSourceDirectory> getGeneratedTestDirectories();
 
+    /**
+     * The set of resource directories.
+     */
+    DomainObjectSet<? extends IdeaSourceDirectory> getResourceDirectories();
+
+    /**
+     * The set of generated resource directories. This is a subset of those directories returned by {@link #getResourceDirectories()}.
+     *
+     * @since 4.5
+     */
+    DomainObjectSet<? extends IdeaSourceDirectory> getGeneratedResourceDirectories();
```

Comment:
Why do we need the `getGeneratedResourceDirectories()`  and `getGeneratedTestResourceDirectories ()` methods? Clients can just test the `isGenerated` attribute for each element.

Reply:
These methods are provided to keep the interface consistent - methods   `getSourceDirectories()`, `getTestDirectories()` also have their counterparts `getGeneratedSourceDirectories()`, getGeneratedTestDirectories().